chronos/asyncmacro2.nim

```diff
@@ -124,41 +116,28 @@ proc asyncSingleProc(prc: NimNode): NimNode {.compileTime.} =
   # ->   complete(chronosInternalRetFuture, result)
   let internalFutureSym = ident "chronosInternalRetFuture"
   var iteratorNameSym = genSym(nskIterator, $prcName)
-  var procBody = prc.body.processBody(internalFutureSym, subtypeIsVoid)
+  var procBody = prc.body.processBody(internalFutureSym)
   # don't do anything with forward bodies (empty)
   if procBody.kind != nnkEmpty:
-    if subtypeIsVoid:
-      let resultTemplate = quote do:
-        template result: auto {.used.} =
-          {.fatal: "You should not reference the `result` variable inside" &
-                   " a void async proc".}
-      procBody = newStmtList(resultTemplate, procBody)
-
     # fix #13899, `defer` should not escape its original scope
     procBody = newStmtList(newTree(nnkBlockStmt, newEmptyNode(), procBody))
 
-    if not subtypeIsVoid:
-      procBody.insert(0, newNimNode(nnkPragma).add(newIdentNode("push"),
-        newNimNode(nnkExprColonExpr).add(newNimNode(nnkBracketExpr).add(
-          newIdentNode("warning"), newIdentNode("resultshadowed")),
-        newIdentNode("off")))) # -> {.push warning[resultshadowed]: off.}
-
-      procBody.insert(1, newNimNode(nnkVarSection, prc.body).add(
-        newIdentDefs(newIdentNode("result"), baseType))) # -> var result: T
-
-      procBody.insert(2, newNimNode(nnkPragma).add(
-        newIdentNode("pop"))) # -> {.pop.})
+    procBody.insert(0): quote do:
```

Comment:
using quote do loses original source location from `newNimNode` unfortunately

Reply:
I think in this case its fine, since this one doesn't contain any newNimNode?

Reply:
```
      procBody.insert(1, newNimNode(nnkVarSection, prc.body).add(
        newIdentDefs(newIdentNode("result"), baseType))) # -> var result: T
```

`var result` in `quote do`?

Reply:
Ohh ok, you meant in the previous code. That is true, but I don't see how this bit could fail (and thus show the invalid lines)

Reply:
Ohh ok, you meant in the previous code. That is true, but I don't see how this bit could fail (and thus show the invalid lines)

Reply:
Ohh ok, you meant in the previous code. That is true, but I don't see how this bit could fail (and thus show the invalid lines)